packages/back-end/src/util/informationSchemas.ts

```diff
@@ -17,17 +17,26 @@ type RowType = {
 export function getPath(dataSourceType: DataSourceType, path: RowType): string {
   const pathArray = Object.values(path);
   const returnValue = pathArray.join(".");
+
   switch (dataSourceType) {
-    // MySQL only supports path's that go two levels deep. E.G. If the full path is database.schema.table.column, it only supports table.column.
+    // MySQL and ClickHouse both support paths that go two levels deep
+    // Backticks help avoid issues with reserved words or special characters
     case "mysql":
+    case "clickhouse":
       if (pathArray.length === 1) {
         return "";
       } else {
-        return pathArray.slice(1, pathArray.length).join(".");
+        return pathArray
+          .slice(1)
+          .map((part) => "`" + part + "`") // Wrap each path part in backticks for safety
+          .join(".");
       }
 
     case "bigquery":
-      return "`" + returnValue + "`"; // BigQuery requires backticks around the path
+      return "`" + returnValue + "`"; // BigQuery requires backticks around the full path
+    case "growthbook_clickhouse":
+      return pathArray[pathArray.length - 1];
```

Comment:
This whole function is very fragile the way it is written.  It depends on the order of object keys, which is bad.  This `growthbook_clickhouse` case doesn't handle column paths, only table paths.

It needs a rewrite and a suite of unit tests.

```ts
expect(
  getPath("growthbook_clickhouse", {
    tableCatalog: "catalog";
    tableSchema: "schema";
    tableName: "table";
    columnName: "column";
  })
).toBe("table.column");
```

Reply:
I'm working through a refactor right now.

@jdorn Looking through it - when building a informationSchema we're calling `getPath` at the `db` level, the `schema` level, and the `table` level, but we're only using the path for tables. 

We're also building the path for each column in an `InformationSchemaTable` record.

I cant think of any uses for the path outside of the table. We currently use it to power the double click feature in the SchemaBrowser. and it'll eventually be used in the auto-complete suggestion feature.

Is there a reason im not seeing?  If not, we could deprecate the path for all but the `tables` and it could vastly simplify the logic of get path.

Reply:
Yeah, I think that's fine. I'd also be in favor of not storing the path in Mongo and generating it on demand when we need it.

Reply:
Yea, I think that is the ideal solution - eliminates the need for orgs to refresh their schemas/tables to get the new path. I think that + only generating the path for tables has greatly reduced the complexity.

Just pushed up changes that do both of those tasks (move the generation to the front end and eliminates the path for databases, schemas, and columns, and just builds it for tables.